querybook/server/logic/elasticsearch.py

```diff
@@ -78,13 +80,19 @@ def get_hosted_es():
     return hosted_es
 
 
+def _get_partial_dict(field_to_generator, fields=None):
+    if fields is None:
```

Comment:
```py
fields = fields or field_to_generator.keys()
return {field: field_to_generator[field]() for field in fields if field in field_to_generator}

Reply:
Slightly changed to the following to check whether getter is a callable or value:

```
fields = fields or field_to_getter.keys()
return {
    field: getter() if callable(getter) else getter
    for field, getter in field_to_getter.items()
    if field in fields
}
```

Reply:
super minor, but
```
for field, getter in field_to_getter.items()
    if field in fields
``` 
is n^2, so its faster to check `if field in field_to_getter`

Reply:
ooh good point, fixed that!